### PR TITLE
Force timeit to not capture stdout

### PR DESCRIPTION
**Pull request by WindSoilder**
# Description
Fixes:  #11996

After this change `let t = timeit ^ls` will list current directory to stdout.
```
❯ let t = timeit ^ls
CODE_OF_CONDUCT.md      Cargo.lock              Cross.toml              README.md               aaa                     benches                 devdocs                 here11                  scripts                 target                  toolkit.nu              wix
CONTRIBUTING.md         Cargo.toml              LICENSE                 a.txt                   assets                  crates                  docker                  rust-toolchain.toml     src                     tests                   typos.toml
```

If user don't want such behavior, he can redirect the stdout to `std null-stream` easily
```
> use std
> let t = timeit { ^ls o> (std null-device) }
```

# User-Facing Changes
NaN

# Tests + Formatting
Done

# After Submitting
Nan

### CODE REVIEW

File: crates/nu-command/src/debug/timeit.rs [hunk 1]
```diff
@@ -44,14 +44,16 @@ impl Command for TimeIt {
         // Get the start time after all other computation has been done.
         let start_time = Instant::now();
 
+        // reset outdest, so the command can write to stdout and stderr.
+        let mut eval_stack = stack.clone().reset_out_dest();
         if let Some(command_to_run) = command_to_run {
             if let Some(block_id) = command_to_run.as_block() {
```
Comment (IanManske) [inline crates/nu-command/src/debug/timeit.rs:50]:
```suggestion
            let stack = &mut stack.push_redirection(None, None);
            if let Some(block_id) = command_to_run.as_block() {
```

Comment (IanManske) [reply crates/nu-command/src/debug/timeit.rs:50]: I didn't document the stack redirection API too well, but a providing `None` to `push_redirection` will clear the corresponding pipe redirection if there is one. So, with this change, I don't think you need any of the other changes in this file. (E.g., this avoids the `Stack` clone.)

**Review by IanManske — COMMENTED** [review]
Thanks, this makes sense / looks good to me!

File: crates/nu-command/src/debug/timeit.rs (hunk 1 above)
Comment (WindSoilder) [reply crates/nu-command/src/debug/timeit.rs:50]: Oh thanks!  After looking into documentation, `push_redirection` makes more sence to me

**Pull request merged**